specs/core/1_new_shards.md

```diff
@@ -0,0 +1,546 @@
+
+<!-- /TOC -->
+
+## Introduction
+
+This document describes the shard transition function (data layer only) and the shard fork choice rule as part of Phase 1 of Ethereum 2.0.
+
+## Configuration
+
+### Misc
+
+| Name | Value | Unit | Duration |
+| - | - | - | - | 
+| `MAX_SHARDS` | `2**10` (= 1024) |
+| `ACTIVE_SHARDS` | `2**6` (= 64) |
+| `ONLINE_PERIOD` | `2**3` (= 8) | epochs | ~51 min |
+| `LIGHT_CLIENT_COMMITTEE_SIZE` | `2**7` (= 128) |
+| `LIGHT_CLIENT_COMMITTEE_PERIOD` | `2**8` (= 256) | epochs | ~27 hours |
+| `SHARD_BLOCK_CHUNK_SIZE` | `2**18` (= 262,144) | |
+| `MAX_SHARD_BLOCK_CHUNKS` | `2**2` (= 4) | |
+| `BLOCK_SIZE_TARGET` | `3 * 2**16` (= 196,608) | |
+| `SHARD_BLOCK_OFFSETS` | `[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233]` | |
+| `MAX_SHARD_BLOCKS_PER_ATTESTATION` | `len(SHARD_BLOCK_OFFSETS)` | |
+| `EMPTY_CHUNK_ROOT` | `hash_tree_root(BytesN[SHARD_BLOCK_CHUNK_SIZE]())` | |
+| `MAX_GASPRICE` | `2**14` (= 16,384) | Gwei | |
+| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `2**3` (= 8) | |
+| `DOMAIN_SHARD_LIGHT_CLIENT` | `192` | |
+| `DOMAIN_SHARD_PROPOSAL` | `193` | |
+
+## Containers
+
+### `ShardBlockWrapper`
+
+```python
+class ShardBlockWrapper(Container):
+    shard_parent_root: Hash
+    beacon_parent_root: Hash
+    slot: Slot
+    body: BytesN[SHARD_BLOCK_CHUNK_SIZE]
+    signature: BLSSignature
+```
+
+### `ShardSignedHeader`
+
+```python
+class ShardSignedHeader(Container):
+    shard_parent_root: Hash
+    beacon_parent_root: Hash
+    slot: Slot
+    body_root: Hash
+```
+
+### `ShardState`
+
+```python
+class ShardState(Container):
+    slot: Slot
+    gasprice: Gwei
+    root: Hash
+    latest_block_hash: Hash
+```
+
+### `AttestationData`
+
+```python
+class AttestationData(Container):
+    slot: Slot
+    index: CommitteeIndex
+    # LMD GHOST vote
+    beacon_block_root: Hash
+    # FFG vote
+    source: Checkpoint
+    target: Checkpoint
+    # Shard transition root
+    shard_transition_root: Hash
+```
+
+### `ShardTransition`
+
+```python
+class ShardTransition(Container):
+    # Starting from slot
+    start_slot: Slot
+    # Shard block lengths
+    shard_block_lengths: List[uint64, MAX_SHARD_BLOCKS_PER_ATTESTATION]
+    # Shard data roots
+    shard_data_roots: List[List[Hash, MAX_SHARD_BLOCK_CHUNKS], MAX_SHARD_BLOCKS_PER_ATTESTATION]
+    # Intermediate state roots
+    shard_state_roots: List[ShardState, MAX_SHARD_BLOCKS_PER_ATTESTATION]
+    # Proposer signature aggregate
+    proposer_signature_aggregate: BLSSignature
+```
+
+### `Attestation`
+
+```python
+class Attestation(Container):
+    aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
+    data: AttestationData
+    custody_bits: List[Bitlist[MAX_VALIDATORS_PER_COMMITTEE], MAX_SHARD_BLOCKS_PER_ATTESTATION]
+    signature: BLSSignature
+```
+
+### `IndexedAttestation`
+
+```python
+class IndexedAttestation(Container):
+    committee: List[ValidatorIndex, MAX_VALIDATORS_PER_COMMITTEE]
+    attestation: Attestation
```

Comment:
It seems very different from [v09x `IndexedAttestation`](https://github.com/ethereum/eth2.0-specs/blob/v09x/specs/core/0_beacon-chain.md#indexedattestation), maybe give it a new name?